server/src/main/scala/org/scassandra/server/actors/ProtocolActor.scala

```diff
@@ -15,70 +15,77 @@
  */
 package org.scassandra.server.actors
 
-import akka.actor.{Actor, ActorLogging, ActorRef}
+import akka.actor.{Actor, ActorLogging, ActorRef, ActorSystem}
 import org.scassandra.codec.Consistency.Consistency
 import org.scassandra.codec._
 import org.scassandra.codec.datatype.DataType
-import org.scassandra.server.priming.query.{Fatal, Prime, Reply}
+import org.scassandra.server.actors.priming.PrimeQueryStoreActor.{Fatal, Prime, Reply}
 
 trait ProtocolActor extends Actor with ActorLogging {
 
-  def write(message: Message, requestHeader: FrameHeader, recipient: Option[ActorRef] = None) = {
-    val target = recipient.getOrElse(sender)
+  def write(message: Message, requestHeader: FrameHeader, target: ActorRef): Unit = {
```

Comment:
Definitely a good change to make the target explicit instead of inferring sender 👍

Reply:
Yeah i started to call this in call backs from futures so didn't want to accidentally use the sender as it is mutable